mysql 8.0.22 -> 8.0.25

###### Checklist

<!-- Note that these are hard requirements -->

- [X] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)
- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: mysql80

current version: 8.0.22
desired version: 8.0.25

###### Notify maintainers

maintainers: Orivej Desh <orivej@gmx.fr>

###### Note for maintainers

Please tag this issue in your PR.

I have started to work on this but the derivation build fails.

```
[ 95%] Building CXX object storage/innobase/CMakeFiles/innobase.dir/fts/fts0plugin.cc.o
[ 95%] Building CXX object storage/innobase/CMakeFiles/innobase.dir/handler/ha_innodb.cc.o
[ 95%] Building CXX object storage/innobase/CMakeFiles/innobase.dir/handler/ha_innopart.cc.o
[ 95%] Building CXX object storage/innobase/CMakeFiles/innobase.dir/handler/handler0alter.cc.o
[ 95%] Building CXX object storage/innobase/CMakeFiles/innobase.dir/handler/i_s.cc.o
In file included from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h: In function 'const char* google::protobuf::io::VarintParse(const char*, T*)':
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:177:12: error: expected unqualified-id before '=' token
  177 |     T byte = static_cast<uint8>(p[i]);
      |            ^
In file included from /build/mysql-8.0.25/storage/innobase/include/fsp0fsp.h:36,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:33:
/build/mysql-8.0.25/storage/innobase/include/univ.i:387:14: error: expected primary-expression before 'unsigned'
  387 | #define byte unsigned char
      |              ^~~~~~~~
In file included from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:178:11: error: expected ';' before 'unsigned'
  178 |     res += byte << (i * 7);
      |           ^
      |           ;
In file included from /nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:140,
                 from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/build/mysql-8.0.25/storage/innobase/include/univ.i:387:14: error: expected primary-expression before 'unsigned'
  387 | #define byte unsigned char
      |              ^~~~~~~~
/build/mysql-8.0.25/storage/innobase/include/univ.i:387:14: error: expected ')' before 'unsigned'
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:180:9: note: to match this '('
  180 |     if (PROTOBUF_PREDICT_TRUE(byte < 128)) {
      |         ^~~~~~~~~~~~~~~~~~~~~
In file included from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:184:31: error: expected ')' before ';' token
  184 |     extra += 128ull << (i * 7);
      |                               ^
      |                               )
In file included from /nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:140,
                 from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:180:9: note: to match this '('
  180 |     if (PROTOBUF_PREDICT_TRUE(byte < 128)) {
      |         ^~~~~~~~~~~~~~~~~~~~~
In file included from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:184:31: error: expected ')' before ';' token
  184 |     extra += 128ull << (i * 7);
      |                               ^
      |                               )
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:180:8: note: to match this '('
  180 |     if (PROTOBUF_PREDICT_TRUE(byte < 128)) {
      |        ^
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h: In function 'const char* google::protobuf::io::ReadSize(const char*, google::protobuf::int32*)':
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:201:17: error: expected unqualified-id before '=' token
  201 |     uint32 byte = static_cast<uint8>(p[i]);
      |                 ^
In file included from /build/mysql-8.0.25/storage/innobase/include/fsp0fsp.h:36,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:33:
/build/mysql-8.0.25/storage/innobase/include/univ.i:387:14: error: expected primary-expression before 'unsigned'
  387 | #define byte unsigned char
      |              ^~~~~~~~
In file included from /nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:140,
                 from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/build/mysql-8.0.25/storage/innobase/include/univ.i:387:14: error: expected primary-expression before 'unsigned'
  387 | #define byte unsigned char
      |              ^~~~~~~~
/build/mysql-8.0.25/storage/innobase/include/univ.i:387:14: error: expected ')' before 'unsigned'
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:204:9: note: to match this '('
  204 |     if (PROTOBUF_PREDICT_TRUE(byte < 128)) {
      |         ^~~~~~~~~~~~~~~~~~~~~
In file included from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:208:31: error: expected ')' before ';' token
  208 |     extra += 128ull << (i * 7);
      |                               ^
      |                               )
In file included from /nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:140,
                 from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:204:9: note: to match this '('
  204 |     if (PROTOBUF_PREDICT_TRUE(byte < 128)) {
      |         ^~~~~~~~~~~~~~~~~~~~~
In file included from /build/mysql-8.0.25/sql/immutable_string.h:49,
                 from /build/mysql-8.0.25/sql/hash_join_buffer.h:72,
                 from /build/mysql-8.0.25/sql/composite_iterators.h:53,
                 from /build/mysql-8.0.25/sql/sql_lex.h:63,
                 from /build/mysql-8.0.25/sql/thd_raii.h:39,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:73:
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:208:31: error: expected ')' before ';' token
  208 |     extra += 128ull << (i * 7);
      |                               ^
      |                               )
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:204:8: note: to match this '('
  204 |     if (PROTOBUF_PREDICT_TRUE(byte < 128)) {
      |        ^
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:210:15: error: expected unqualified-id before '=' token
  210 |   uint32 byte = static_cast<uint8>(p[4]);
      |               ^
/nix/store/lrrwa3q23c6v8b6lw4d253y3bjv7g21m-protobuf-3.7.1/include/google/protobuf/io/coded_stream.h:212:12: error: expected unqualified-id before '>=' token
  212 |   if (byte >= 8) return nullptr;
      |            ^~
In file included from /build/mysql-8.0.25/storage/innobase/include/fsp0fsp.h:36,
                 from /build/mysql-8.0.25/storage/innobase/fsp/fsp0fsp.cc:33:
/build/mysql-8.0.25/storage/innobase/include/univ.i:387:14: error: expected primary-expression before 'unsigned'
  387 | #define byte unsigned char
      |              ^~~~~~~~
```

Any ideas?

--- a/pkgs/servers/sql/mysql/8.0.x.nix
+++ b/pkgs/servers/sql/mysql/8.0.x.nix
@@ -1,16 +1,16 @@
 { lib, stdenv, fetchurl, bison, cmake, pkg-config
 , boost, icu, libedit, libevent, lz4, ncurses, openssl, protobuf, re2, readline, zlib, zstd
-, numactl, perl, cctools, CoreServices, developer_cmds, libtirpc, rpcsvc-proto
+, numactl, perl, cctools, CoreServices, developer_cmds, libtirpc, rpcsvc-proto, curl
 }:
 
 let
 self = stdenv.mkDerivation rec {
   pname = "mysql";
-  version = "8.0.22";
+  version = "8.0.25";
 
   src = fetchurl {
     url = "https://dev.mysql.com/get/Downloads/MySQL-${self.mysqlVersion}/${pname}-${version}.tar.gz";
-    sha256 = "9fd85bb243940ef8234d21384ef421a0962fd4d13406fc1420efa902115ce17a";
+    sha256 = "c16aa9cf621bc028efba2bb11f3c36a323b125fa0d108ff92fab60e46309206e";
   };
 
   patches = [
@@ -27,7 +27,7 @@ self = stdenv.mkDerivation rec {
   '';
 
   buildInputs = [
-    boost icu libedit libevent lz4 ncurses openssl protobuf re2 readline zlib
+    boost curl icu libedit libevent lz4 ncurses openssl protobuf re2 readline zlib
     zstd
   ] ++ lib.optionals stdenv.isLinux [
     numactl libtirpc

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -19838,7 +19838,7 @@ in
     inherit (darwin) cctools developer_cmds;
     inherit (darwin.apple_sdk.frameworks) CoreServices;
     boost = boost173; # Configure checks for specific version.
-    protobuf = protobuf3_7;
+    protobuf = protobuf3_11;
     icu =  icu67;
   };
 

